src/order/conditionally_complete_lattice.lean

```diff
@@ -642,51 +664,90 @@ open_locale classical
+      refine cSup_le _ _,
+      { refine mt _ h,
+        rw ne_empty_iff_exists_mem at hs,
+        intro h2,
+        rcases hs with ⟨⟨⟩|b, hb⟩, assumption,
+        exfalso, revert h2, exact ne_empty_of_mem hb},
+      { intros a ha, exact some_le_some.1 (hb ha)}},
+    { rintro (⟨⟩|b) hb, exact _root_.le_refl _,
+      exfalso, apply h_1, use b, intros a ha, exact some_le_some.1 (hb ha)}}
 end
 
-lemma has_glb (s : set (with_top α)) : ∃a, is_glb s a :=
+lemma is_lub_Sup (s : set (with_top α)) : is_lub s (Sup s) :=
 begin
-  by_cases hs : ∃x:α, ↑x ∈ s,
-  { rcases hs with ⟨x, hxs⟩,
-    refine ⟨(Inf {a : α | ↑a ∈ s} : α), _, _⟩,
-    exact (assume a has, (coe_le_iff _ _).2 $ assume x hx, cInf_le (bdd_below_bot _) $
-      show ↑x ∈ s, from hx ▸ has),
-    { assume a has,
-      rcases (le_coe_iff _ _).1 (has hxs) with ⟨x, rfl, h⟩,
-      exact (coe_le_coe.2 $ le_cInf (ne_empty_of_mem hxs) $
-        assume b hbs, coe_le_coe.1 $ has hbs) } },
-  exact ⟨⊤, assume a, match a with
-    | some a, ha := (hs ⟨a, ha⟩).elim
-    | none,   ha := _root_.le_refl _
-    end,
-    assume a _, le_top⟩
+  by_cases hs : s = ∅,
+  { rw hs,
+    show is_lub ∅ (ite _ _ _),
+    split_ifs,
+    { cases h},
+    { rw [preimage_empty, cSup_empty], exact is_lub_empty},
+    { exfalso, apply h_1, use ⊥, rintro a ⟨⟩}
+  },
+  exact is_lub_Sup' hs,
 end
 
-noncomputable instance : has_Sup (with_top α) := ⟨λs, classical.some $ has_lub s⟩
-noncomputable instance : has_Inf (with_top α) := ⟨λs, classical.some $ has_glb s⟩
+/-- The Inf of a bounded-below set is its greatest lower bound for a conditionally
+complete lattice with a top. -/
+lemma is_glb_Inf' {β : Type*} [conditionally_complete_lattice β]
+  {s : set (with_top β)} (hs : bdd_below s) : is_glb s (Inf s) :=
+begin
+  split,
+  { show ite _ _ _ ∈ _,
+    split_ifs,
+    { intros a ha, exact top_le_iff.2 (set.mem_singleton_iff.1 (h ha))},
+    { rintro (⟨⟩|a) ha, exact le_top,
+      refine some_le_some.2 (cInf_le _ ha),
+      rcases hs with ⟨⟨⟩|b, hb⟩,
+      { exfalso,
+        apply h,
+        intros c hc,
+        rw [mem_singleton_iff, ←top_le_iff],
+        exact hb hc},
+      use b,
+      intros c hc,
+      refine some_le_some.1 (hb hc)}},
+  { show ite _ _ _ ∈ _,
+    split_ifs,
+    { intros _ _, exact le_top},
+    { rintro (⟨⟩|a) ha,
+      { exfalso, apply h, intros b hb, exact set.mem_singleton_iff.2 (top_le_iff.1 (ha hb))},
+      { refine some_le_some.2 (le_cInf _ _),
+        { refine mt _ h,
+          rintro hs (⟨⟩|b) hb,
+          { exact mem_singleton _},
+          { exfalso, apply not_mem_empty b, rwa ←hs}},
+        { intros b hb,
+          rw ←some_le_some,
+          refine ha hb}}}}
+end
 
-lemma is_lub_Sup (s : set (with_top α)) : is_lub s (Sup s) := classical.some_spec _
-lemma is_glb_Inf (s : set (with_top α)) : is_glb s (Inf s) := classical.some_spec _
+lemma is_glb_Inf (s : set (with_top α)) : is_glb s (Inf s) :=
+begin
+  by_cases hs : bdd_below s,
+  { exact is_glb_Inf' hs},
```

Comment:
```suggestion
  { exact is_glb_Inf' hs },
```

Reply:
I didn't know this convention. There are many other places where I don't leave a space before a `}`. I can maybe use a regexp...

Reply:
I am confused about this. Am I not allowed `{X : Type*}`?

Reply:
Oh this is just for tactics? There are lots of regex hits for \S\} in tactic proof and many were not done by me. I've changed them all!